src/lib.rs

```diff
@@ -420,26 +417,23 @@ impl Builder {
         // Construct background task
         let (sender, mut receiver) = mpsc::channel::<Bytes>(buffer);
 
-        let bg_task = Box::pin(
-            async move {
-                // Reconnection loop
-                loop {
-                    // Do a DNS lookup if `addr` is a hostname
-                    let addrs = lookup_host(&addr).await.into_iter().flatten();
-
-                    // Loop through the IP addresses that the hostname resolved to
-                    for addr in addrs {
-                        if let Err(_err) = handle_udp_connection(addr, &mut receiver).await {
-                            // TODO: Add handler
-                        }
-                    }
+        let bg_task = Box::pin(async move {
+            // Reconnection loop
+            loop {
+                // Do a DNS lookup if `addr` is a hostname
+                let addrs = lookup_host(&addr).await.into_iter().flatten();
 
-                    // Sleep before re-attempting
-                    time::sleep(time::Duration::from_millis(timeout_ms as u64)).await;
+                // Loop through the IP addresses that the hostname resolved to
+                for resolved in addrs {
+                    if let Err(err) = handle_udp_connection(resolved, &mut receiver).await {
+                        tracing::error!(error = %err, %resolved, "can't connect to graylog TCP backend");
+                    }
                 }
+
+                // Sleep before re-attempting
+                time::sleep(time::Duration::from_millis(timeout_ms as u64)).await;
             }
-            .with_subscriber(NoSubscriber),
```

Comment:
The motivation for the `with_subscriber` here is https://github.com/hlb8122/tracing-gelf/pull/16. Do you have some confirmation that the explosion of logging is, without supression around the entirity of the background task, still suppressed? By the looks of the original issue the explosion occurs within the actual socket send so the `NoSubscriber` on `send_all`/`send` is sufficient?

Reply:
Yes, I can confirm that this logging feedback still doesn't happen on the TCP connection. And it kinda makes sense, as well, the only thing that new log messages cause is that the loop/`send_all` is executed, so only stuff in there can create a feedback loop.